### PR TITLE
[T3A4][T10-A4]KelvinNg

### PR DESCRIPTION
**Pull request by kelvinnpc**
Referred [T3A4][T09-A2]Clarence Chee for how to check for deleted file

### CODE REVIEW

**Comment by kelvinnpc** [comment]
Ready for Review

File: src/seedu/addressbook/storage/StorageFile.java [hunk 1]
```diff
@@ -85,13 +85,21 @@ private static boolean isValidPath(Path filePath) {
      */
     public void save(AddressBook addressBook) throws StorageOperationException {
 
-        /* Note: Note the 'try with resource' statement below.
+    	checkStorageFile();    	
+    	createAndWriteFile(addressBook);
```
Comment (nirandiw) [inline src/seedu/addressbook/storage/StorageFile.java]: Good idea to extract the method. Increases SLAP.

File: src/seedu/addressbook/storage/StorageFile.java [hunk 2]
```diff
@@ -101,7 +109,16 @@ public void save(AddressBook addressBook) throws StorageOperationException {
...
             throw new StorageOperationException("Error converting address book into storage format");
         }
-    }
+	}
+
+	/*
+	 * Check to see if storage file was deleted
+	 */
+    private void checkStorageFile() throws StorageOperationException {
+		if(!path.toFile().exists()) {
```
Comment (nirandiw) [inline src/seedu/addressbook/storage/StorageFile.java]: - Good try. But may be throw the exception after you recovered the file.

- Where did you handle this exception? Best place would be the executeCommand() Main.java

File: src/seedu/addressbook/storage/StorageFile.java [hunk 3]
```diff
@@ -85,13 +85,21 @@ private static boolean isValidPath(Path filePath) {
...
+     */
+	private void createAndWriteFile(AddressBook addressBook) throws StorageOperationException {
+		/* Note: Note the 'try with resource' statement below.
          * More info: https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html
          */
         try (final Writer fileWriter =
                      new BufferedWriter(new FileWriter(path.toFile()))) {
 
-            final AdaptedAddressBook toSave = new AdaptedAddressBook(addressBook);
+        	final AdaptedAddressBook toSave = new AdaptedAddressBook(addressBook);
```
Comment (nirandiw) [inline src/seedu/addressbook/storage/StorageFile.java]: Badd indentation

**Comment by nirandiw** [comment]
Some comments added. Please close the PR after reading comments

**Pull request closed without merging**